[FLINK-15279][doc] Document new executeAsync() method and the newly introduced JobClient (#10684)
apache/flink: docs/dev/api_concepts.md

> The `execute()` method is returning a `JobExecutionResult`, this contains execution

Reviewer: ```suggestion
The `execute()` method will wait for the job to finish and then return  a `JobExecutionResult`, this contains execution
```

times and accumulator results.

> If you don't want to wait for the job finished, you are able to trigger the program execution by calling

Reviewer: ```suggestion
If you don't want to wait for the job to finish, you can trigger asynchronous job execution by calling
```

`executeAysnc()` on the `StreamExecutionEnvironment`. It will return a `JobClient` with which you can

> communicate with the job you just submitted. For instances, here is how to implemenet the same semantic

Reviewer: ```suggestion
communicate with the job you just submitted. For instance, here is how to implement the semantics
```
